### PR TITLE
Send payment_method_type analytic when you attempt to create a PM, co…

### PR DESCRIPTION
**Pull request by yuki-stripe**
…nfirm a PI, confirm an SI

## Summary
Create a PM, confirm a PI, confirm an SI all send `payment_method_type` now.
Confirm PI no longer sends `bancontact` as `source_type` when there is no source.

## Motivation
https://jira.corp.stripe.com/browse/IOS-1610

## Testing
Ran basic integration example, created a PM and confirmed a PI.  Saw the expected analytics.

### CODE REVIEW

File: Stripe/STPAPIClient.m [hunk 1]
```diff
@@ -847,14 +846,13 @@ - (void)createPaymentMethodWithParams:(STPPaymentMethodParams *)paymentMethodPar
...
     NSCAssert(paymentMethodParams != nil, @"'paymentMethodParams' is required to create a PaymentMethod");
     NSCAssert(paymentMethodParams.rawTypeString != nil, @"Set the `type` or `rawTypeString` property on paymentMethodParams.");
+    [[STPAnalyticsClient sharedClient] logPaymentMethodCreationAttemptWithConfiguration:self.configuration paymentMethodType:paymentMethodParams.rawTypeString];
+    
     [STPAPIRequest<STPPaymentMethod *> postWithAPIClient:self
                                                endpoint:APIEndpointPaymentMethods
                                              parameters:[STPFormEncoder dictionaryForObject:paymentMethodParams]
                                            deserializer:[STPPaymentMethod new]
                                              completion:^(STPPaymentMethod *paymentMethod, __unused NSHTTPURLResponse *response, NSError *error) {
-                                                 if (error == nil && paymentMethod != nil) {
```
Comment (yuki-stripe) [inline Stripe/STPAPIClient.m]: I don't know why this didn't match the other analytics, which fire regardless of success.

File: Stripe/STPAnalyticsClient.m [hunk 2]
```diff
@@ -252,7 +252,7 @@ - (void)logSetupIntentConfirmationAttemptWithConfiguration:(STPPaymentConfigurat
     NSMutableDictionary *payload = [self.class commonPayload];
     [payload addEntriesFromDictionary:@{
                                         @"event": @"stripeios.setup_intent_confirmation",
-                                        @"payment_method_type": paymentMethodType ?: @"unknown",
+                                        @"source_type": paymentMethodType ?: @"unknown",
```
Comment (yuki-stripe) [inline Stripe/STPAnalyticsClient.m]: deliberately overloading `source_type` here like android does.

Comment (mshafrir-stripe) [reply Stripe/STPAnalyticsClient.m]: We could also pass this as `payment_method_type` from the client and figure out what to do with it on the backend. For this PR maybe simpler to go with what you have.

File: Stripe/STPAPIClient.m (hunk 1 above)
Comment (yuki-stripe) [reply Stripe/STPAPIClient.m]: @csabol-stripe Can you confirm I'm not clobbering something intentional here?

File: Stripe/STPAnalyticsClient.m [hunk 3]
```diff
@@ -218,13 +218,13 @@ - (void)logSourceCreationAttemptWithConfiguration:(STPPaymentConfiguration *)con
...
-- (void)logPaymentMethodCreationSucceededWithConfiguration:(STPPaymentConfiguration *)configuration
-                                           paymentMethodID:(NSString *)paymentMethodID {
+- (void)logPaymentMethodCreationAttemptWithConfiguration:(STPPaymentConfiguration *)configuration
+                                       paymentMethodType:(NSString *)paymentMethodType {
     NSDictionary *configurationDictionary = [self.class serializeConfiguration:configuration];
     NSMutableDictionary *payload = [self.class commonPayload];
     [payload addEntriesFromDictionary:@{
                                         @"event": @"stripeios.payment_method_creation",
-                                        @"payment_method_id": paymentMethodID,
+                                        @"source_type": paymentMethodType,
```
Comment (yuki-stripe) [inline Stripe/STPAnalyticsClient.m]: @csabol-stripe mind looking at this one too?

**Pull request merged**